src/java/org/apache/cassandra/index/sai/plan/Operation.java

```diff
@@ -207,7 +209,38 @@ private static int getPriority(org.apache.cassandra.cql3.Operator op)
 
     static RangeIterator buildIterator(QueryController controller)
     {
-        return Node.buildTree(controller.filterOperation()).analyzeTree(controller).rangeIterator(controller);
+        var orderings = controller.filterOperation().expressions()
```

Comment:
I do like that we used to have the "is this an ANN query" logic encapsulated-ish

Reply:
Would you prefer moving this to the QueryController? If (when) we differentiate order operations from filter operations, I think we will store those order ops in this class, which is why it seems acceptable to keep this here for now.

Reply:
cleaning it up when we split out order from other expressions wfm